Common++/src/IpAddress.cpp

```diff
@@ -246,10 +244,8 @@ bool IPv6Address::operator!=(const IPv6Address& other)
 
 IPv6Address& IPv6Address::operator=(const IPv6Address& other)
 {
-	if (m_pInAddr != NULL)
-		delete m_pInAddr;
-
-	m_pInAddr = (in6_addr*)new uint8_t[sizeof(in6_addr)];
+	if (m_pInAddr == NULL)
+		m_pInAddr = (in6_addr*)new uint8_t[sizeof(in6_addr)];
```

Comment:
I'm not sure why you did this change. Could you please explain?

Reply:
The buffer `m_pInAddr` have the constant size `sizeof(in6_addr)`. The copy operation is made with the same size: `memcpy(m_pInAddr, other.m_pInAddr, sizeof(in6_addr))`. So if a pointer to buffer is not null we can just copy the data into it. It eliminates, at least, delete operator.
Besides, we can remove the condition `if (m_pInAddr == NULL)` because the buffer is always allocated by constructors.

Reply:
More precisely, we can remove following code at all:
```
if (m_pInAddr == NULL)
  m_pInAddr = (in6_addr*)new uint8_t[sizeof(in6_addr)];

```